src/pages/index.js

```diff
@@ -2,88 +2,219 @@
 profileFormClass.setEventListeners();
 cardAddFormClass.setEventListeners();
+pictureFormClass.setEventListeners();
+confirmFormClass.setEventListeners();
 
-editFormValidator.enableValidation();
-addFormValidator.enableValidation();
+enableValidation(validationSettings);
 
 // FUNCTIONS //
 function handleImageClick(card) {
   cardPreviewClass.open({ name: card._name, link: card._link });
 }
 
-function handleCardAddSubmit(data) {
-  renderCard({ name: data.title, link: data.link });
-  cardAddFormClass.close();
-  cardAddForm.reset();
+function handleCardAddSubmit(inputData) {
+  cardAddFormClass.renderLoading(true);
+  api
+    .addCard(inputData)
+    .then((data) => {
+      const { _id, name, link, isLiked } = data;
+      const newCardData = { _id, name, link, isLiked };
+      cardSection.addItem(createCard(newCardData));
+      formValidators[cardAddForm.getAttribute("name")].disableBtn();
+      cardAddForm.reset();
+      cardAddFormClass.close();
+    })
+    .catch((err) => {
+      console.error(`Error: failed to add card. ${err}`);
+    })
+    .finally(() => {
+      cardAddFormClass.renderLoading(false);
+    });
 }
 
 function handleProfileFormSubmit(data) {
-  profileInfoClass.setUserInfo(data);
-  profileFormClass.close();
+  profileFormClass.renderLoading(true);
+  api
+    .setUserInfo(data)
+    .then(() => {
+      profileInfoClass.setUserInfo({ name: data.name, about: data.about });
+      formValidators[profileEditForm.getAttribute("name")].disableBtn();
+      profileFormClass.close();
+    })
+    .catch((err) => {
+      console.error(`Error handling profile form submit. ${err}`);
+    })
+    .finally(() => {
+      profileFormClass.renderLoading(false);
+    });
 }
 
-function renderCard(card) {
-  const cardElementClass = new Card({
-    name: card.name,
-    link: card.link,
-    cardSelector: "#card-template",
-    handleImageClick: handleImageClick,
+function handlePictureSubmit(avatarUrl) {
+  pictureFormClass.renderLoading(true);
+  api
+    .changeAvatar(avatarUrl)
+    .then(() => {
+      profileInfoClass.setUserInfo({ avatar: avatarUrl.picture });
+      formValidators[profilePictureForm.getAttribute("name")].disableBtn();
+      pictureFormClass.close();
+    })
+    .catch(console.error)
+    .finally(() => {
+      pictureFormClass.renderLoading(false);
+    });
+}
+
+function handleLikeBtn(card) {
+  api
+    .setCardLiked(card.id, card.getLikedStatus())
+    .then((updatedCard) => {
+      card.handleIsLiked(updatedCard.isLiked);
+    })
+    .catch(console.error);
+}
+
+function handleDeleteBtn(card) {
+  confirmFormClass.open();
+  confirmFormClass.handleConfirm(() => {
+    confirmFormClass.viewLoading(true);
+    api
+      .removeCard(card.id)
+      .then(() => {
+        card.removeCard();
+        confirmFormClass.close();
+      })
+      .catch((err) => {
+        console.error(`Failed to delete card. ${err}`);
+      })
+      .finally(confirmFormClass.viewLoading(false));
```

Comment:
`      .finally(confirmFormClass.viewLoading(false));
`

In blocks `then`, `catch`, `finally` you need to pass only the **reference** to the function (not a function call) otherwise the function is called immediately without waiting for the response from the server. You can pass an **unnamed** arrow function as well. 



```js
      .finally(() => confirmFormClass.viewLoading(false));


```